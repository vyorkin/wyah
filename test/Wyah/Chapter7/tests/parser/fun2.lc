-- Functions 2

let twice f x = f (f x);
let on g f = \x y -> g (f x) (f y);
let ap f x = f (f x);
