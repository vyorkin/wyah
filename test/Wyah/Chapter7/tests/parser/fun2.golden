Program [Decl "ap" (ELam (Var "x") (ELam (Var "f") (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EVar (Var "x")))))),Decl "on" (ELam (Var "f") (ELam (Var "g") (ELam (Var "x") (ELam (Var "y") (EApp (EApp (EVar (Var "g")) (EApp (EVar (Var "f")) (EVar (Var "x")))) (EApp (EVar (Var "f")) (EVar (Var "y")))))))),Decl "twice" (ELam (Var "x") (ELam (Var "f") (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EVar (Var "x"))))))]